internal/query/current_sequence.go

```diff
@@ -17,6 +18,66 @@ type LatestSequence struct {
 	Timestamp time.Time
 }
 
+type CurrentSequences struct {
+	SearchResponse
+	CurrentSequences []*CurrentSequence
+}
+
+type CurrentSequence struct {
+	ProjectionName  string
+	AggregateType   string
+	CurrentSequence uint64
+	Timestamp       time.Time
+}
+
+type CurrentSequencesSearchQueries struct {
+	SearchRequest
+	Queries []SearchQuery
+}
+
+func (q *Queries) SearchCurrentSequences(ctx context.Context, queries *CurrentSequencesSearchQueries) (failedEvents *CurrentSequences, err error) {
+	query, scan := prepareCurrentSequencesQuery()
+	stmt, args, err := queries.toQuery(query).ToSql()
+	if err != nil {
+		return nil, errors.ThrowInvalidArgument(err, "QUERY-MmFef", "Errors.Query.InvalidRequest")
+	}
+
+	rows, err := q.client.QueryContext(ctx, stmt, args...)
+	if err != nil {
+		return nil, errors.ThrowInternal(err, "QUERY-22H8f", "Errors.Internal")
+	}
+	return scan(rows)
+}
+
+func (q *Queries) latestSequence(ctx context.Context, projection table) (*LatestSequence, error) {
+	query, scan := prepareLatestSequence()
+	stmt, args, err := query.Where(sq.Eq{
+		CurrentSequenceColProjectionName.identifier(): projection.name,
+	}).ToSql()
+	if err != nil {
+		return nil, errors.ThrowInternal(err, "QUERY-5CfX9", "Errors.Query.SQLStatement")
+	}
+
+	row := q.client.QueryRowContext(ctx, stmt, args...)
+	return scan(row)
+}
+
+func (q *Queries) ClearCurrentSequence(ctx context.Context, projectionName string) (err error) {
+	tx, err := q.client.Begin()
+	if err != nil {
+		return errors.ThrowInternal(err, "QUERY-9iOpr", "Errors.RemoveFailed")
+	}
+	_, err = tx.Exec(fmt.Sprintf("TRUNCATE %s", projectionName))
+	if err != nil {
+		return errors.ThrowInternal(err, "QUERY-3n92f", "Errors.RemoveFailed")
+	}
+	_, err = tx.Exec(fmt.Sprintf("UPDATE %s SET (%s) = ($1) WHERE (%s = $2)", currentSequencesTable, CurrentSequenceColCurrentSequence, CurrentSequenceColProjectionName), 0, projectionName)
```

Comment:
why are some parameters passed to the `fmt.Sprintf` and others as arguments to the exec function?
I would fmt.Sprintf the statement on start and then pass the necessary values and set `0` directly in the string anyway, as it's always this value

Reply:
I prefer the current solution. Sql injection is still a thing

Reply:
right, but isn't then the exec right before an even bigger issue:
```
_, err = tx.Exec(fmt.Sprintf("TRUNCATE %s", projectionName))
```
https://github.com/caos/zitadel/pull/2770/files/fd5577d48160eda602f8284625548f62cfc21641#diff-f51caeb4ea903bf7a8c3e07f1b820e5938c93fd17e6fd109d2c01757ed8509b4R70

Reply:
IMO not because we control the variable names, everything from outside zitadel is a potential risk IMO